larpandoracontent/LArMonitoring/VisualParticleMonitoringAlgorithm.cc

```diff
@@ -0,0 +1,428 @@
+/**
+ *  @file   larpandoracontent/LArMonitoring/VisualParticleMonitoringAlgorithm.cc
+ *
+ *  @brief  Implementation of the particle visualisation algorithm.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+
+#include "larpandoracontent/LArMonitoring/VisualParticleMonitoringAlgorithm.h"
+
+#include "larpandoracontent/LArHelpers/LArPfoHelper.h"
+#include "larpandoracontent/LArObjects/LArCaloHit.h"
+#include "larpandoracontent/LArObjects/LArMCParticle.h"
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+VisualParticleMonitoringAlgorithm::VisualParticleMonitoringAlgorithm() :
+    m_visualizeMC(false),
+    m_visualizePfo(false),
+    m_groupMCByPdg(false),
+    m_showPfoByPid(false),
+    m_showPfoMatchedMC(false),
+    m_isTestBeam{false}
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+VisualParticleMonitoringAlgorithm::~VisualParticleMonitoringAlgorithm()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode VisualParticleMonitoringAlgorithm::Run()
+{
+#ifdef MONITORING
```

Comment:
Looks like this should all be good, but a test with MONITORING=OFF always helpful (and we could add this to our travis CI build matrix).

Reply:
Yes, this has been tested with MONITORING=OFF and it works as expected